File: Mar-18/13-03-18/java/02-Monk-and-Lucky-Minimum.java
import java.util.*;


class TestClass {
    public static void main(String args[] ) throws Exception {
        Scanner input = new Scanner(System.in);
        
        int t=input.nextInt();
        while(t-->0)
        {
            int n=input.nextInt();
            int mini=Integer.MAX_VALUE,cnt=0;
            
            for(int i=0;i<n;i++)
            {
                int val=input.nextInt();
                if(val<mini)
                {
                    mini=val;cnt=1;
                }
                else if(val==mini)
                    cnt++;
            }
        
        if(cnt%2==0)
            System.out.println("Unlucky");
        else
            System.out.println("Lucky");
    
        }        
    }
}

File: Mar-18/13-03-18/python/02-Monk-and-Lucky-Minimum.py
import sys

for _ in range(int(input())):
    n = int(input())
    mini = sys.maxsize
    cnt = 0
    for i in input().split():
        i = int(i)
        if(i<mini):
            mini = i
            cnt = 1
        elif(i==mini):
            cnt += 1
    if(cnt%2==0):
        print("Unlucky")
    else:
        print("Lucky")
